app/models/mentor.rb

```diff
@@ -17,7 +17,7 @@ class Mentor < ApplicationRecord
   has_many :mentor_memberships
   has_many :schools, through: :mentor_memberships
 
-  has_many :placement_mentor_joins, dependent: :destroy
+  has_many :placement_mentor_joins, dependent: :restrict_with_error
   has_many :placements, through: :placement_mentor_joins
```

Comment:
These two lines look like they should belong in a `Placements::Mentor` record.

Reply:
The `Mentor → Placements` association is not new. It has existed for ~4 months since #334, so I'd consider a refactor of that to be outside the scope of this PR.